### PR TITLE
fix types for hooks

### PR DESCRIPTION
**Pull request by gautamsi**
I found soem inconsistencies in the afterDelete and beforeDelete hook types.

found missing `updatedItem` in the hook for afterChange and beforeChange hook

### CODE REVIEW

File: packages-next/types/src/config/hooks.ts [hunk 1]
```diff
@@ -92,7 +92,7 @@ type ValidateInputHook<TGeneratedListTypes extends BaseGeneratedListTypes> = (
 ) => Promise<void> | void;
 
 type BeforeOrAfterChangeHook<TGeneratedListTypes extends BaseGeneratedListTypes> = (
-  args: ArgsForCreateOrUpdateOperation<TGeneratedListTypes>
+  args: ArgsForCreateOrUpdateOperation<TGeneratedListTypes> & ({ updatedItem: TGeneratedListTypes['inputs']['update']})
```
Comment (gautamsi) [inline packages-next/types/src/config/hooks.ts:95]: should this be properly added with `TGeneratedListTypes['inputs']['create']` as well?

**Comment by gautamsi** [comment]
@timleslie also found that there is no typings for authentication hooks

File: packages-next/types/src/config/hooks.ts (hunk 1 above)
Comment (timleslie) [reply packages-next/types/src/config/hooks.ts:95]: Yeah, I think this should be:

```
updatedItem: TGeneratedListTypes['inputs']['create'] | TGeneratedListTypes['inputs']['update'];
```

**Review by timleslie — CHANGES_REQUESTED** [review]
Good catches, just one change as per comments below 👍

File: packages-next/types/src/config/hooks.ts (hunk 1 above)
Comment (gautamsi) [reply packages-next/types/src/config/hooks.ts]: done

**Review by timleslie — APPROVED** [review]
👍

**Pull request merged**